services/app-api/src/authn/thirdPartyAuthn.ts

```diff
@@ -0,0 +1,27 @@
+import type { Result } from 'neverthrow'
+import { ok, err } from 'neverthrow'
+import type { UserType } from '../domain-models/index'
+import type { Store } from '../postgres'
+import { lookupUserAurora } from './cognitoAuthn'
+
+export async function userFromThirdPartyAuthorizer(
+    store: Store,
+    userId: string
+): Promise<Result<UserType, Error | undefined>> {
+    try {
+        // Lookup user from postgres
+        const auroraUser = await lookupUserAurora(store, userId)
+        if (auroraUser instanceof Error) {
+            return err(auroraUser)
+        }
+
+        if (auroraUser === undefined) {
+            return err(auroraUser)
+        }
+
+        return ok(auroraUser)
+    } catch (e) {
+        console.error('ERROR: failed to look user in postgres')
```

Comment:
I _think_ this should be `recordException` and going through OTEL. Assuming we would want to alert on this? Also noticing we don't have technical docs on OTEL besides the the initial ADR... so maybe you haven't seen this code yet @pearl-truss

Reply:
@haworku this error gets returned with the function and handled [here](https://github.com/Enterprise-CMCS/managed-care-review/blob/main/services/app-api/src/handlers/apollo_gql.ts#L86)  and is a part of the [tracing middleware](https://github.com/Enterprise-CMCS/managed-care-review/blob/main/services/app-api/src/handlers/apollo_gql.ts#L131C10-L131C27) that is using an otel tracer. I'm not particularly familiar with this area of the code base. Given the above, do you think `recordException` should be used here?

Reply:
Yeah, unfortunately we need to do those `recordExceptions` anywhere we want to record a span to a trace. Unlike pretty much every other monitoring system I've used, OTEL doesn't do this stuff automatically for you (there is some auto-instrumentation, but most of the time you need to do the things yourself). We should write up some docs about these patterns and clean things up a bit, it's not all that intuitive!

Reply:
I'm not 100% sure what needs to happen here but I took an educated guess and pushed that code up. Lmk if anything needs changing @haworku @mojotalantikite

Reply:
I'll merge since I'm seeing things working but if anything comes up for improving this code I can make those changes in https://github.com/Enterprise-CMCS/managed-care-review/pull/2195